internal/configs/transportserver_test.go

```diff
@@ -62,6 +62,164 @@ func TestTransportServerExString(t *testing.T) {
+			StatusZone:               "tcp-listener",
+			ProxyPass:                "ts_default_tcp-server_tcp-app",
+			Name:                     "tcp-server",
+			Namespace:                "default",
+			ProxyConnectTimeout:      "60s",
+			ProxyNextUpstream:        false,
+			ProxyNextUpstreamTries:   0,
+			ProxyNextUpstreamTimeout: "0s",
+			ProxyTimeout:             "10m",
+			HealthCheck:              nil,
+			Snippets:                 []string{"deny  192.168.1.1;", "allow 192.168.1.0/24;"},
+		},
+	}
+	tsc := &transportServerConfigurator{
+		isPlus:         true,
+		enableSnippets: true,
+	}
+	result := generateTransportServerConfig(&transportServerEx, listenerPort, tsc)
+	if !reflect.DeepEqual(result, expected) {
+		t.Errorf("generateTransportServerConfig() returned \n%+v but expected \n%+v", result, expected)
+	}
+}
+
+func TestGenerateTransportServerConfigForTCPSnippetsNotEnabled(t *testing.T) {
+	transportServerEx := TransportServerEx{
+		TransportServer: &conf_v1alpha1.TransportServer{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name:      "tcp-server",
+				Namespace: "default",
+			},
+			Spec: conf_v1alpha1.TransportServerSpec{
+				Listener: conf_v1alpha1.TransportServerListener{
+					Name:     "tcp-listener",
+					Protocol: "TCP",
+				},
+				Upstreams: []conf_v1alpha1.Upstream{
+					{
+						Name:    "tcp-app",
+						Service: "tcp-app-svc",
+						Port:    5001,
+					},
+				},
+				Action: &conf_v1alpha1.Action{
+					Pass: "tcp-app",
+				},
+				Snippets: "deny  192.168.1.1;\nallow 192.168.1.0/24;",
+			},
+		},
+		Endpoints: map[string][]string{
+			"default/tcp-app-svc:5001": {
+				"10.0.0.20:5001",
+			},
+		},
+	}
+
+	listenerPort := 2020
+
+	expected := version2.TransportServerConfig{
+		Upstreams: []version2.StreamUpstream{
+			{
+				Name: "ts_default_tcp-server_tcp-app",
+				Servers: []version2.StreamUpstreamServer{
+					{
+						Address:     "10.0.0.20:5001",
+						MaxFails:    1,
+						FailTimeout: "10s",
+					},
+				},
+				UpstreamLabels: version2.UpstreamLabels{
+					ResourceName:      "tcp-server",
+					ResourceType:      "transportserver",
+					ResourceNamespace: "default",
+					Service:           "tcp-app-svc",
+				},
+			},
+		},
+		Server: version2.StreamServer{
+			Port:                     2020,
+			UDP:                      false,
+			StatusZone:               "tcp-listener",
+			ProxyPass:                "ts_default_tcp-server_tcp-app",
+			Name:                     "tcp-server",
+			Namespace:                "default",
+			ProxyConnectTimeout:      "60s",
+			ProxyNextUpstream:        false,
+			ProxyNextUpstreamTries:   0,
+			ProxyNextUpstreamTimeout: "0s",
+			ProxyTimeout:             "10m",
+			Snippets:                 []string{},
+		},
+	}
+	tsc := &transportServerConfigurator{
+		isPlus:         true,
+		enableSnippets: false,
+		warnings:       newWarnings(),
+	}
+	result := generateTransportServerConfig(&transportServerEx, listenerPort, tsc)
+	if !reflect.DeepEqual(result, expected) {
```

Comment:
we need to check that warnings are generated in this case. for example, like we do here https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/configs/virtualserver_test.go#L3341

Reply:
We confined Warnings to `generateTransportServerConfig()` as the supporting infrastructure isn't available. Warnings are handled at a higher level and the TransportServer status is updated. Waiting on https://github.com/nginxinc/kubernetes-ingress/pull/1415 for this.

Reply:
they are confined, but not tested - generateTransportServerConfig doesn't check that they were generated? Could you add a test?